### DISCUSSION TITLE
problem with delete and undo

### DISCUSSION THREAD

**Discussion by Ndmanh123** [clientIO/joint#1688] [Q&A]
Hi im try to delete a cell and undo it but it don't work like i want here is my code

```js
paper.on("cell:pointerup", function (cellView) {
  self.selectedElement = cellView.model;
}
document.addEventListener("keyup", function (evt) {
  if (!this.selectedElement && !this.selection) return;
  if (evt.code === "Backspace" || evt.code === "Delete") {
    this.selectedElement.remove();
    this.selectedElement = null;
   }
}
```

I have 2 cell when I delete 1 cell and undo it just keep repeat remove and add that cell .when i press undo is should show that cell i delete again and then delete the cell in the order that i render it not repeat add and delete that same cell Im doing vuejs

**Comment by Ndmanh123** [comment]
My undo and redo work well before i delete a cell i have check the undo and redo stack it look like the delete cell just keep been add over and over in two stack any one know please help thank

**Reply by kumilingus** [reply]
What is the value of `commandManager.undoStack` and `commandManager.redoStack` after you delete the cell?

**Reply by Ndmanh123** [reply]
it keep repeat  the same cell at the end of commandManager.undoStack and redoStack so it after you delete the cell and undo it show again than if you keep undo more time it just add and remove that same cell over and over

**Reply by kumilingus** [reply]
I think you have to provide a code that reproduces the issue. Perhaps, you have an event handler in place that triggers an action in response to a graph change (the undo action triggers a change again).
